backend/benefit/calculator/rules.py

```diff
@@ -157,70 +170,77 @@ def get_maximum_monthly_pay_subsidy(self):
 
-            if date_ranges[0].pay_subsidy:
+        for sub_range in date_ranges:
+            if len(date_ranges) > 1:
                 self._create_row(
-                    DescriptionRow,
-                    description_fi_template="Vähennettävät korvaukset / kk",
+                    DateRangeDescriptionRow,
+                    start_date=sub_range.start_date,
+                    end_date=sub_range.end_date,
+                    prefix_text="Ajalta",
                 )
-                pay_subsidy_monthly_eur = self._create_row(
-                    PaySubsidyMonthlyRow,
-                    pay_subsidy=date_ranges[0].pay_subsidy,
-                    max_subsidy=self.get_maximum_monthly_pay_subsidy(),
-                ).amount
-            else:
-                pay_subsidy_monthly_eur = 0
+            monthly_deductions = self.create_deduction_rows(sub_range)
+
             self._create_row(
                 SalaryBenefitMonthlyRow,
                 max_benefit=self.SALARY_BENEFIT_MAX,
-                pay_subsidy_monthly_eur=pay_subsidy_monthly_eur,
+                monthly_deductions=monthly_deductions,
+            )
+            self._create_row(
+                SalaryBenefitSubTotalRow,
+                start_date=sub_range.start_date,
+                end_date=sub_range.end_date,
             )
-            self._create_row(SalaryBenefitTotalRow)
-        else:
-            self._create_row(SalaryCostsRow)
-            self._create_row(StateAidMaxMonthlyRow)
-            for start_date, end_date, pay_subsidy in date_ranges:
-                self._create_row(
-                    DateRangeDescriptionRow,
-                    start_date=start_date,
-                    end_date=end_date,
-                    prefix_text="Ajalta",
-                )
-                if pay_subsidy:
-                    self._create_row(
-                        DescriptionRow,
-                        description_fi_template="Vähennettävät korvaukset / kk",
-                    )
-                    pay_subsidy_monthly_eur = self._create_row(
-                        PaySubsidyMonthlyRow,
-                        pay_subsidy=pay_subsidy,
-                        max_subsidy=self.get_maximum_monthly_pay_subsidy(),
-                    ).amount
-                else:
-                    pay_subsidy_monthly_eur = 0
-
-                self._create_row(
-                    SalaryBenefitMonthlyRow,
-                    max_benefit=self.SALARY_BENEFIT_MAX,
-                    pay_subsidy_monthly_eur=pay_subsidy_monthly_eur,
-                )
-                self._create_row(
-                    SalaryBenefitSubTotalRow, start_date=start_date, end_date=end_date
-                )
 
+        if len(date_ranges) > 1:
             self._create_row(
                 DateRangeDescriptionRow,
-                start_date=start_date,
-                end_date=end_date,
+                start_date=sub_range.start_date,
```

Comment:
This looks quite weird because `sub_range` is a local variable inside the for-loop. Is this supposed to be correct? Or should it be `date_ranges[0].start_date` and `date_ranges[-1].end_date`?

Reply:
The goal is that, if the the calculation is done separately for multiple date ranges, to add for each time range an
"Ajalta dd.mm.yyyy - dd.mm.yyyy" row. In the UI design, page "palkan - kahdella ajalla", you can see the two separate rows

![image](https://user-images.githubusercontent.com/156631/144489549-4a5c14be-6f13-437e-9dab-dfd477f79d2d.png)

from:
https://futuricedesigners.invisionapp.com/console/share/T734X5SPR2/813764061

But if there is only one time range in the calculation, then the row "Ajalta dd.mm.yyyy - dd.mm.yyyy" is redundant, that's why the row is created only if len(date_ranges) > 1

Reply:
Oops, my previous comment was wrong, there's indeed a bug in the calculation of the "Koko ajalta" row, fixed